### PR TITLE
[Merged by Bors] - perf: squeeze some really slow `simp`s

### PR DESCRIPTION
**Pull request by grunweg**
See the individual commit messages for details: each simp was particular slow, and is much faster now.

---

I'm happy to revert particular instances you don't agree with.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Good idea: I created a tracking issue and referenced it in the changed places.

File: Mathlib/Algebra/GroupWithZero/Pointwise/Finset.lean [hunk 1]
```diff
@@ -185,12 +185,16 @@ open scoped RightActions
...
+      MulOpposite.op_inv, inv_eq_zero, MulOpposite.op_eq_zero_iff, inv_inv,
+      MulOpposite.smul_eq_mul_unop, MulOpposite.unop_op, mul_inv_rev, ha]
 
 @[simp] lemma inv_op_smul_finset_distrib₀ (a : α) (s : Finset α) : (s <• a)⁻¹ = a⁻¹ • s⁻¹ := by
   obtain rfl | ha := eq_or_ne a 0
   · obtain rfl | hs := s.eq_empty_or_nonempty <;> simp [*]
-  · ext; simp [← inv_smul_mem_iff₀, *]
+  · ext; simp only [mem_inv', ne_eq, MulOpposite.op_eq_zero_iff, not_false_eq_true, ←
+      inv_smul_mem_iff₀, MulOpposite.smul_eq_mul_unop, MulOpposite.unop_inv, MulOpposite.unop_op,
+      inv_eq_zero, inv_inv, smul_eq_mul, mul_inv_rev, ha]
```
Comment (b-mehta) [inline Mathlib/Algebra/GroupWithZero/Pointwise/Finset.lean]: These ones being slow surprises me! Do we know what was making them so slow?

Comment (grunweg) [reply Mathlib/Algebra/GroupWithZero/Pointwise/Finset.lean]: I don't! Feel free to investigate (I don't really work in this area...) and fix the underlying issue.

**Review by Ruben-VandeVelde — APPROVED** [review]
Investigation can proceed in parallel, I think. 

maintainer merge

**Comment by b-mehta** [comment]
Agreed! Thanks for this.

bors merge

**Pull request closed without merging**